File: gcc/testsuite/gcc.dg/gomp/appendix-a/a.33.1.c
/* { dg-do compile } */
/* { dg-require-effective-target tls } */

#include <stdio.h>
float x, y;
#pragma omp threadprivate(x, y)
void
init (float a, float b)
{
#pragma omp single copyprivate(a,b,x,y)
  {
    scanf ("%f %f %f %f", &a, &b, &x, &y);
  }
}

File: gcc/testsuite/gfortran.dg/gomp/appendix-a/a.33.1.f90
! { dg-do compile }
! { dg-require-effective-target tls_native }

      SUBROUTINE INIT(A,B)
      REAL A, B
        COMMON /XY/ X,Y
!$OMP THREADPRIVATE (/XY/)
!$OMP SINGLE
          READ (11) A,B,X,Y
!$OMP END SINGLE COPYPRIVATE (A,B,/XY/)
      END SUBROUTINE INIT
